### PR TITLE
Update react-day-picker typings for version 1.2.0

### PR DESCRIPTION
**Pull request by jkillian**

### CODE REVIEW

**Comment by JKillian** [comment]
:+1:

File: react-day-picker/react-day-picker.d.ts [hunk 1]
```diff
@@ -13,18 +13,28 @@ declare module "react-day-picker" {
...
 
-    interface Props extends __React.Props<DayPicker>{
+    interface CaptionElementProps extends React.Props<any> {
+        date?: Date;
+        localeUtils?: LocaleUtils;
+        locale?: string;
+        onClick?: React.MouseEventHandler;
+    }
+
+    interface Props extends React.Props<DayPicker>{
```
Comment (adidahiya) [inline react-day-picker/react-day-picker.d.ts]: space before `{`

File: react-day-picker/react-day-picker-tests.tsx [hunk 2]
```diff
@@ -20,3 +20,17 @@ function MyComponent() {
...
 DayPicker2.DateUtils.isDayInRange(new Date(), { from: new Date() });
+
+// test interface for captionElement prop
+interface MyCaptionProps extends ReactDayPicker.CaptionElementProps { }
+class Caption extends React.Component<MyCaptionProps, {}> {
+    render() {
+        const { date, locale, localeUtils, onClick } = this.props;
+        return (
+            <div className="DayPicker-Caption" onClick={ onClick }>
+              { localeUtils.formatMonthTitle(date, locale) }
```
Comment (adidahiya) [inline react-day-picker/react-day-picker-tests.tsx]: 4-space indentation

**Comment by adidahiya** [comment]
minor nits but lgtm

**Comment by adidahiya** [comment]
hey @horiuchi can we get this merged? it looks fine as it is :+1:

**Comment by horiuchi** [comment]
@JKillian Thanks mate!
@adidahiya Thanks for review!

**Pull request merged**